lib/rom/commands/result.rb

```diff
@@ -24,6 +24,20 @@ def to_ary
       end
       alias_method :to_a, :to_ary
 
+      # Return true if command successful
+      #
+      # @api public
+      def success?
+        !value.nil? && error.nil?
+      end
+
+      # Return true if command failed
+      #
+      # @api public
+      def failure?
+        value.nil? && !error.nil?
```

Comment:
Interestingly isn't the same... but possibly more correct. 
What if both are nil or both are present?